lib/usecases/login_use_case.dart

```diff
@@ -21,14 +22,11 @@ class LoginUseCase extends UseCase<AuthToken, LoginCredential> {
   const LoginUseCase(this._repository);
 
   @override
-  Future<Result<AuthToken>> call(LoginCredential credential) async {
-    try {
-      final value =
-          await _repository.login(credential.email, credential.password);
-      return Success(value);
-    } catch (e) {
-      final apiError = ApiError(NetworkExceptions.getDioException(e), e);
-      return Future.value(Failed(apiError));
-    }
+  Future<Result<AuthToken>> call(LoginCredential credential) {
+    return _repository
+        .login((credential.email), credential.password)
+        .then<Result<AuthToken>>((value) => Success(value))
+        .onError<DioError>((error, stackTrace) => Failed(
+            UseCaseError(NetworkExceptions.getDioException(error), error)));
```

Comment:
I suggest renaming 
`NetworkExceptions.getDioException(error), error)` -> `NetworkExceptions.fromDioException(error), error)`
will describe more correctly what this function will do 👍

Reply:
how about this @sleepylee 😁

Reply:
ohhh `get` -> `from` 🤔  not a bad idea, let me update it 🙆‍♂️